### ISSUE TITLE
docs: Gateway implementation guide for managed vs. unmanaged gateway lifecycles and intra vs extra cluster

### ISSUE THREAD

**Issue by shaneutt** [kubernetes-sigs/gateway-api#892]
**What would you like to be added**:

This is a request for documentation for implementations which would provide additional guidance and prescription on how to implement `Gateway` controllers for different kinds of underlying Gateway infrastructure options. Particularly the guide should cover:

- `Gateway`s which are "unmanaged" and refer to an existing deployment

Additional Goals:

- define clear terminology about the underlying lifecycle management and network location of `Gateways` 

**Why this is needed**:

This is a direct follow up from a community meeting where the overall topic was discussed, and was a follow up to:

- @howardjohn's [Document](https://docs.google.com/document/d/1BjmBUaNnDsk8LcDF0YbItV1kRRtc3eiWBdgvRg-1hfo) on the subject
- A related discussion which suggested some implementation details for how to solve some of the problems in this domain with spec changes: https://github.com/kubernetes-sigs/gateway-api/discussions/877

**Comment by shaneutt** [comment]
As per our conversation in zoom today, I'll be taking this one on.

/assign @shaneutt

**Comment by shaneutt** [comment]
while the originally wording of this issue may need some updating as it hasn't aged as well as I would like I still think some guidance on implementation of the "unmanaged" Gateway mode (which in my team's implementation we are supporting in addition to the "managed" mode) is valuable for future developers. I'm not sure it's going to be something I do right away, but if nobody else wants to take this one I will intend to do it as soon as is feasible.

/assign

**Comment by shaneutt** [comment]
I've gone through a couple drafts that I have reviewed with peers on this one. I've always found my own results wanting. The fact of the matter is that trying to implement for `unmanaged` mode is difficult, and it's problematic to provide any prescription for others to do the same since it gets deep into implementation-specific territory for each implementor.

I'm also convinced now that the value of writing this guide is pretty limited since we expect that most implementors are going to be able to implement `Gateway` in a native way, and my team is even considering not releasing the `unmanaged` way. My main suggestion to anyone reading this who might actually be looking for guidance is to ensure their implementation passes the conformance tests. Since I created this, I am going to close it for now. If anyone is looking to work on this I'll be happy to re-open it.
